### ISSUE TITLE
Ui Component File Field

### ISSUE THREAD

**Issue by thomas-kl1** [magento/magento2#38908]
### Preconditions and environment

- Latest Magento version

### Steps to reproduce

Add the following field in a ui component form:

```xml
<file name="file">
            <settings>
                <label translate="true">Select File to Import</label>
                <dataScope>file</dataScope>
                <validation>
                    <rule name="required-entry" xsi:type="boolean">true</rule>
                </validation>
            </settings>
        </file>
```

### Expected result

The form is submitted without error.

### Actual result

Even if we select a file, the validation fails, like no file was selected.
h
![346533395-e1667aa9-ca43-4516-8b18-53967a2fc682](https://github.com/magento/magento2/assets/20971693/fe04b194-d969-42de-8fc2-cfa475efffd1)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Hello @thomas-kl1,

Thank you for the report and collaboration!

We did not see the error while using the fileUploader component:
```
<field name="file">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="required" xsi:type="boolean">true</item>
                    <item name="validation" xsi:type="array">
                        <item name="required-entry" xsi:type="boolean">true</item>
                    </item>
                    <item name="sortOrder" xsi:type="number">333</item>
                    <item name="formElement" xsi:type="string">fileUploader</item>
                    <item name="label" xsi:type="string" translate="true">add file</item>
                </item>
            </argument>
</field>
```

Can you please check and let us know if this is working at your end.

Thank you.

**Comment by thomas-kl1** [comment]
fileUploader and file are two different component.

**Comment by engcom-November** [comment]
It is recommended to use fileUploader, but in case if you are using file component for some specific use case, can you let us know the same.

**Comment by engcom-November** [comment]
Yes file component can minimize the extra work needed for ajax, so confirming this issue.

**Comment by Mohamed-Asar** [comment]
@magento I am working on this
